controllers/logging/elasticsearch_controller.go

```diff
@@ -82,9 +94,37 @@ func (r *ElasticsearchReconciler) Reconcile(request ctrl.Request) (ctrl.Result,
 	return reconcileResult, nil
 }
 
+// handleSecret returns true if metaname is such that (cr_name) matches or that (cr_name + "-proxy") matches
+func handleESSecret(meta metav1.Object) bool {
+	// iterate over registeredKibanas that match the namespace
+	namespace := meta.GetNamespace()
+
+	registeredElasticsearches.mux.Lock()
+	defer registeredElasticsearches.mux.Unlock()
+
+	for _, es := range registeredElasticsearches.registered {
+		if es.Namespace == namespace {
+			if es.Name == meta.GetName() {
+				return true
+			}
+		}
+	}
+
+	return false
+}
+
 func (r *ElasticsearchReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Watch for updates to the elasticsearch cluster secret
+	secretPred := predicate.Funcs{
+		UpdateFunc:  func(e event.UpdateEvent) bool { return handleESSecret(e.MetaNew) },
+		CreateFunc:  func(e event.CreateEvent) bool { return false },
+		DeleteFunc:  func(e event.DeleteEvent) bool { return handleESSecret(e.Meta) },
+		GenericFunc: func(e event.GenericEvent) bool { return false },
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		Named("elasticsearch-controller").
 		For(&loggingv1.Elasticsearch{}).
+		Watches(&source.Kind{Type: &corev1.Secret{}}, &handler.EnqueueRequestForObject{}, builder.WithPredicates(secretPred)).
```

Comment:
Won't this just run the normal reconcile loop in the case where the secret is updated? That isn't what we want to do... if the secret is updated we just need to go over all the nodes for that CR and mark that they will require a cert restart